paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -42,18 +46,30 @@ import java.util.Locale
 class PaparazziPlugin : Plugin<Project> {
   @OptIn(ExperimentalStdlibApi::class)
   override fun apply(project: Project) {
-    require(project.plugins.hasPlugin("com.android.library")) {
-      "The Android Gradle library plugin must be applied before the Paparazzi plugin."
+    var setup = false
+    project.plugins.withId("com.android.library") {
+      setup(project, project.extensions.getByType(LibraryExtension::class.java).libraryVariants)
+      setup = true
     }
+    project.plugins.withId("com.android.application") {
+      setup(project, project.extensions.getByType(AppExtension::class.java).applicationVariants)
+      setup = true
+    }
+    project.afterEvaluate {
+      require(setup) {
+        "The Android Gradle application or library plugin must be applied together with Paparazzi plugin."
+      }
+    }
+  }
```

Comment:
Feels a bit hacky, but couldn't figure out a better way. Is this still necessary? I guess it's good DX to alert if applied to wrong module rather than ignoring, but at the same time it could prevent blanket application with `allprojects` block.

Reply:
https://github.com/cashapp/paparazzi/pull/301/commits/071f255036f502950a75461875dfb601a0bb1a39